### PR TITLE
Add Allegro 2.0

### PR DESCRIPTION
**Pull request by mtekman**
(DeCODE's) linkage analysis program that is the main contender against (Helmholtz-Muenchen's) Genehunter. 

Same functionality, but uses a globalised Viterbi method to resolve haplotypes instead. Marginally more accurate on X-linked analyses too.

### CODE REVIEW

File: tools/allegro/allegro.xml [hunk 1]
```diff
@@ -0,0 +1,229 @@
+<tool id="allegro" name="Allegro" version="2" >
+    <description>Linkage and Haplotype analysis</description>
+    <requirements>
+	<requirement type="package" version="2">allegro</requirement>
+    </requirements>
+    <command detect_errors='exit_code'>allegro $allegro_conf</command>
```
Comment (bebatut) [inline tools/allegro/allegro.xml]: The `CDATA` is missing here (same for the configfile)

File: tools/allegro/allegro.xml [hunk 2]
```diff
@@ -0,0 +1,229 @@
+<tool id="allegro" name="Allegro" version="2" >
+    <description>Linkage and Haplotype analysis</description>
+    <requirements>
+	<requirement type="package" version="2">allegro</requirement>
+    </requirements>
```
Comment (bebatut) [inline tools/allegro/allegro.xml]: Can you add the `version_commands`, please?

File: tools/allegro/allegro.xml [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+PREFILE ${inp_ped}
+DATFILE ${inp_dat}
+
+#if $cond_haplotypes.opt_haplotypes
+HAPLOTYPE haplo.out ${out_ihaplo} ${out_founder} inher.out
+    #if $cond_haplotypes.opt_crossover
+CROSSOVERRATE combined.out ${out_combined_cross}
+    #end if
+#else
+    #if '$cond_haplotypes.section_linkage.cond_linktype.extra_linkage_type.value' == 'Allele Sharing':
```
Comment (bebatut) [inline tools/allegro/allegro.xml]: I am not sure if you need the quotes around the param for the test here

File: tools/allegro/allegro.xml [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+    </configfiles>
+
+    <inputs>
+	<!-- Have to add this format to the list of expected types -->
+	<param name="inp_ped" type="data" format="linkage_pedin" label="Pedigree" />
+	<param name="inp_dat" type="data" format="linkage_recomb" label="Recombination Freqs." />
+	<param name="inp_map" type="data" format="linkage_map" label="Marker Positions" optional="true" />
+
+	<conditional name="cond_haplotypes" >
+	    <param name="opt_haplotypes" type="boolean" value="false" label="Haplotypes" />
```
Comment (bebatut) [inline tools/allegro/allegro.xml]: Can you use a `select` type and not a `boolean` here (part of the best practices)? Check http://galaxy-iuc-standards.readthedocs.io/en/latest/best_practices/tool_xml.html#dynamic-options

Comment (bebatut) [reply tools/allegro/allegro.xml]: Same for the other cases after

File: tools/allegro/allegro.xml [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+            <when value="true" >
+                <param name="opt_crossover" type="boolean" value="false" label="Show crossover rate" />
+            </when>
+            <when value="false" >
+                <section name="section_linkage" title="Linkage" expanded="true" >
+                    <param name="extra_linkage_mptspt" type="boolean" truevalue="mpt" falsevalue="spt" value="true"
+                           label="Use multipoint IBD sharing probabilities, instead of single point" />
+                    <conditional name="cond_linktype" >
+                        <param name="extra_linkage_type" type="select" label="Type of analysis" >
+                            <option value="Allele Sharing" >Allele Sharing</option>
```
Comment (bgruening) [inline tools/allegro/allegro.xml]: @mtekman this is really nitpicking, but can you change the values of options to something that does not have a space inside and preferably all lower-case. For example in your tests you already have it as lower-case.

**Comment by bgruening** [comment]
Thanks @mtekman this comes nicely together!

**Comment by mtekman** [comment]
Getting there - thanks @bgruening  and @bebatut for the reviews!

**Pull request merged**